dask/array/wrap.py

```diff
@@ -146,4 +145,35 @@ def wrap(wrap_func, func, **kwargs):
 ones_like = w_like(np.ones, func_like=np.ones_like)
 zeros_like = w_like(np.zeros, func_like=np.zeros_like)
 empty_like = w_like(np.empty, func_like=np.empty_like)
-full_like = w_like(np.full, func_like=np.full_like)
+
+
+# full and full_like require special casing due to argument check on fill_value
+# Generate wrapped functions only once
+_full = w(np.full)
+_full_like = w_like(np.full, func_like=np.full_like)
+
+
+def full(shape, fill_value, dtype=None, order="C", chunks=None):
+    # np.isscalar has somewhat strange behavior:
+    # https://docs.scipy.org/doc/numpy/reference/generated/numpy.isscalar.html
+    if np.ndim(fill_value) != 0:
+        raise ValueError(f"fill_value must be scalar. Received {fill_value} instead.")
```

Comment:
```suggestion
        raise ValueError(f"fill_value must be scalar. Received {type(fill_value)} instead.")
```

in case `fill_value` has a large string repr?